lib/web_ui/test/compositing_test.dart

```diff
@@ -208,6 +208,35 @@ void main() {
       html.HtmlElement contentAfterReuse = builder2.build().webOnlyRootElement;
       expect(contentAfterReuse.querySelector('canvas').style.zIndex, '-1');
     });
+
+    test('Multiple canvas elements should retain zIndex after update', () async {
+      final SurfaceSceneBuilder builder = SurfaceSceneBuilder();
+      final Picture picture1 = _drawPathImagePath();
+      EngineLayer oldLayer = builder.pushClipRect(
+        const Rect.fromLTRB(10, 10, 300, 300),
+      );
+      builder.addPicture(Offset.zero, picture1);
+      builder.pop();
+
+      html.HtmlElement content = builder.build().webOnlyRootElement;
+      expect(content.querySelector('canvas').style.zIndex, '-2');
+
+      // Force update to scene which will utilize reuse code path.
+      final SurfaceSceneBuilder builder2 = SurfaceSceneBuilder();
+      builder2.pushClipRect(
+          const Rect.fromLTRB(5, 10, 300, 300),
+          oldLayer: oldLayer
+      );
+      final Picture picture2 = _drawPathImagePath();
+      builder2.addPicture(Offset.zero, picture2);
+      builder2.pop();
+
+      html.HtmlElement contentAfterReuse = builder2.build().webOnlyRootElement;
+      List<html.CanvasElement> list =
+          contentAfterReuse.querySelectorAll('canvas');
+      expect(list[0].style.zIndex, '-2');
+      expect(list[1].style.zIndex, '-1');
```

Comment:
If you have two canvases, with z-index -2 and -1, that means that there's an image or text that's between them. Shouldn't the image/text have a z-index that's between the two canvas' z-indexes? The other issue is that if both a canvas and text have negative z-index, then the canvas will appear on top of text due to the Chrome/Safari bug.

Reply:
reverted per offline discussion.